app/controllers/comment_controller.php

```diff
@@ -0,0 +1,32 @@
+<?php
+class CommentController extends AppController
+{
+    public function create()
+    {
+        redirect_guest_user('user/authenticate');
+
+        $thread = Thread::get(Param::get('thread_id'));
+        $page   = Param::get('page_next');
+
+        switch ($page) {
+        case 'create_end':
+            $comment            = new Comment();
+            $comment->body      = Param::get('body');
+            $comment->user_id   = User::getAuthenticated()->id;
```

Comment:
If getAuthenticated returns false, it will cause an error since id is an undefined property